internal/service/resourcegroups/resource.go

```diff
@@ -63,7 +71,11 @@ func resourceResourceCreate(ctx context.Context, d *schema.ResourceData, meta in
 
 	groupARN := d.Get("group_arn").(string)
 	resourceARN := d.Get(names.AttrResourceARN).(string)
-	id := strings.Join([]string{strings.Split(strings.ToLower(groupARN), "/")[1], strings.Split(resourceARN, "/")[1]}, "_")
+	id, err := flex.FlattenResourceId([]string{groupARN, resourceARN}, resourceIDPartCount, false)
```

Comment:
Will this affect existing Terraform resources?
Should add an acceptance test creating in the previous version and refreshing with the updated code just to make sure.

Reply:
Only if they were consuming the `id` output somewhere else in their configuration. In practice I'm not sure how useful this would have been as this was previously a `_` delimited string of group name and some portion of the resource section of the associated resource ARN.

Reply:
I guess add a CHANGELOG entry calling out the `id` format change also?

Reply:
Added test case and additional changelog entry.